resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/tpr/TPRCrudCallsImpl.scala

```diff
@@ -0,0 +1,320 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.kubernetes.tpr
+
+import java.util.concurrent.{ThreadPoolExecutor, TimeUnit}
+
+import io.fabric8.kubernetes.client.{BaseClient, KubernetesClient}
+import okhttp3.{HttpUrl, MediaType, OkHttpClient, Request, RequestBody, Response}
+import okio.{Buffer, BufferedSource}
+import org.json4s.{DefaultFormats, Formats}
+import org.json4s.JsonDSL._
+import org.json4s.jackson.JsonMethods.{compact, parse, pretty, render}
+import org.json4s.jackson.Serialization.{read, write}
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.{blocking, Future, Promise}
+import scala.util.control.Breaks.{break, breakable}
+
+import org.apache.spark.deploy.kubernetes.constants._
+import org.apache.spark.internal.Logging
+import org.apache.spark.SparkException
+import org.apache.spark.util.ThreadUtils
+
+private[spark] trait TPRCrudCalls {
+  /**
+   * This trait contains currently acceptable operations on the SparkJob Resource
+   * CRUD + Watch
+   */
```

Comment:
Javadoc for a trait/class goes on the line immediately prior to the class, not after.  As written, this is javadoc on one of the 5 methods, rather than on the class.